tests/Sulu/Component/Content/SmartContent/Orm/BaseDataProviderTest.php

```diff
@@ -0,0 +1,275 @@
+                2,
+                [['id' => 2], ['id' => 3]],
+                false,
+                [['id' => 2], ['id' => 3]],
+            ],
+            [
+                [],
+                null,
+                2,
+                2,
+                [['id' => 2], ['id' => 3], ['id' => 4]],
+                true,
+                [['id' => 2], ['id' => 3]],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider filtersProvider
+     */
+    public function testResolveFilters(
+        array $filters,
+        $limit,
+        $page,
+        $pageSize,
+        $repositoryResult,
+        $hasNextPage,
+        $items
+    ) {
+        $repository = $this->prophesize(DataProviderRepositoryInterface::class);
+        $repository->findByFilters($filters, $page, $pageSize, $limit)->shouldBeCalled()->willReturn($repositoryResult);
+
+        /** @var BaseDataProvider $provider */
+        $provider = $this->getMockForAbstractClass(BaseDataProvider::class, [$repository->reveal()]);
+
+        $class = new \ReflectionClass(BaseDataProvider::class);
+        $method = $class->getMethod('resolveFilters');
+        $method->setAccessible(true);
+
+        $result = $method->invokeArgs($provider, [$filters, $limit, $page, $pageSize]);
+
+        $this->assertEquals($items, $result[0]);
+        $this->assertEquals($hasNextPage, $result[1]);
+    }
+
+    /**
+     * @dataProvider filtersProvider
+     */
+    public function testResolveDataItems(
+        array $filters,
+        $limit,
+        $page,
+        $pageSize,
+        $repositoryResult,
+        $hasNextPage,
+        $items
+    ) {
+        $repository = $this->prophesize(DataProviderRepositoryInterface::class);
+        $repository->findByFilters($filters, $page, $pageSize, $limit)->shouldBeCalled()->willReturn($repositoryResult);
+
+        /** @var BaseDataProvider $provider */
+        $provider = $this->getMockForAbstractClass(BaseDataProvider::class, [$repository->reveal()]);
+        $provider->expects($this->any())->method('decorateDataItems')->willReturn($items);
+
+        $result = $provider->resolveDataItems(
+            $filters,
+            [],
+            ['locale' => 'en', 'webspace' => 'sulu_io'],
+            $limit,
+            $page,
+            $pageSize
+        );
+
+        $this->assertInstanceOf(DataProviderResult::class, $result);
+        $this->assertEquals($hasNextPage, $result->getHasNextPage());
+        $this->assertEquals($items, $result->getItems());
+        $this->assertEquals([], $result->getReferencedUuids());
+    }
+
+    /**
+     * @dataProvider filtersProvider
+     */
+    public function testResolveResourceItems(
+        array $filters,
+        $limit,
+        $page,
+        $pageSize,
+        $repositoryResult,
+        $hasNextPage,
+        $items
+    ) {
+        $mockedItems = array_map(
+            function ($item) {
+                $mock = $this->prophesize(TestInterface::class);
```

Comment:
Then why don't you use your `ResourceInterface` instead of defining a new one in this test? Isn't it matching here?